src/mem.rs

```diff
@@ -58,3 +62,112 @@ pub unsafe extern "C" fn memcmp(s1: *const u8, s2: *const u8, n: usize) -> i32 {
     }
     0
 }
+
+unsafe fn memcpy_element_unordered_atomic<T: Copy>(dest: *mut T, src: *const T, bytes: usize) {
+    let n = unchecked_div(bytes, mem::size_of::<T>());
```

Comment:
I suppose that Bad Things will happen if this is called with a zero-sized T (which could happen by accident in generic code). Is it the responsibility of the caller to refrain from doing that?

Reply:
These functions are only exposed via the monomorphic __llvm_* wrappers, which ensure that `T` is a primitive unsigned integer type.

Reply:
Ah, thanks, I missed that.